packages/components/src/internal/components/editable/EditableGridPanelForUpdate.tsx

```diff
@@ -33,54 +35,35 @@ interface Props {
     selectionData: Map<string, any>;
     setIsDirty?: (isDirty: boolean) => void;
     singularNoun?: string;
-    updateRows: (schemaQuery: SchemaQuery, rows: any[]) => Promise<any>;
+    updateRows: (schemaQuery: SchemaQuery, rows: any[], comment: string) => Promise<any>;
 }
 
-interface State {
-    dataModels: QueryModel[];
-    editorModels: EditorModel[];
-    error: string;
-    isSubmitting: boolean;
-    loaders: EditableGridLoader[];
-}
-
-export class EditableGridPanelForUpdate extends React.Component<Props, State> {
-    static defaultProps = {
-        singularNoun: 'row',
-        pluralNoun: 'rows',
-    };
-
-    constructor(props: Props) {
-        super(props);
-        const id = props.loader.id;
-
-        this.state = {
-            isSubmitting: false,
-            dataModels: [new QueryModel({ id, schemaQuery: props.queryModel.schemaQuery })],
-            editorModels: [new EditorModel({ id })],
-            error: undefined,
-            loaders: [props.loader],
-        };
-    }
-
-    componentDidMount(): void {
-        this.initEditorModel();
-    }
-
-    initEditorModel = async (): Promise<void> => {
-        const { dataModels, editorModels } = await initEditableGridModels(
-            this.state.dataModels,
-            this.state.editorModels,
-            this.state.loaders,
-            this.props.queryModel
-        );
-        this.setState({ dataModels, editorModels });
-    };
-
-    onGridChange: EditableGridChange = (event, editorModelChanges, dataKeys, data, index = 0): void => {
-        this.setState(state => {
-            const { dataModels, editorModels } = state;
-            return applyEditableGridChangesToModels(
+export const EditableGridPanelForUpdate: FC<Props> = props => {
+    const { containerFilter, onCancel, singularNoun, pluralNoun, ...editableGridProps } = props;
+    const { idField, loader, queryModel, selectionData, getIsDirty, setIsDirty, updateRows, onComplete, } = editableGridProps;
+    const id = loader.id;
+
+    const [dataModels, setDataModels] = useState<QueryModel[]>([new QueryModel({ id, schemaQuery: props.queryModel.schemaQuery })]);
```

Comment:
Rather than new'ing up these objects each render cycle you can pass a function which will only be called on initialization.

```tsx
const [dataModels, setDataModels] = useState<QueryModel[]>(() => [
    new QueryModel({ id, schemaQuery: queryModel.schemaQuery }),
]);
const [editorModels, setEditorModels] = useState<EditorModel[]>(() => [new EditorModel({ id })]);
```

Reply:
Interesting. I didn't know that trick.